### PR TITLE
Included annotations to the "tfc_agent_service_account"

### PR DESCRIPTION
**Pull request by gowripadmajas**
Included annotations to the "tfc_agent_service_account" , so there can be an optional option provided for scenario's to associate IAM roles for service accounts or other annotations for K8's Service account. @redeux : please review this request

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -102,3 +102,8 @@ variable "tfc_url" {
   description = "The Terraform Cloud endpoint.  Must be changed if using Terraform Enterprise."
 }
 
+variable "annotations" {
```
Comment (redeux) [inline variables.tf]: I think this might be better as `service_account_annotations` since it only applies there.

File: variables.tf [hunk 2]
```diff
@@ -102,3 +102,8 @@ variable "tfc_url" {
...
+  description = "Add annotations for service accounts"
```
Comment (redeux) [inline variables.tf]: It would be good to mention that irsa use case in the description

**Comment by redeux** [comment]
Thank you for your contribution! I made a couple of suggestions. Please update the README documentation as well.

File: variables.tf (hunk 2 above)
Comment (gowripadmajas) [reply variables.tf]: Changes incorporated as suggested to the README, variable name and description . Please review @redeux .

**Comment by redeux** [comment]
I’m currently traveling with my family and without my computer, so I’ll have take a look at this early next week when I’m back home.

**Comment by gowripadmajas** [comment]
Hello @redeux , can merge this PR :)

**Comment by niroz89** [comment]
@redeux is this good to be merged ?

**Comment by gowripadmajas** [comment]
@redeux , Creating a new PR with same code snippets but from a different branch, as we need to enhance our current main.Closing this PR .
cc: @lawliet @niroz89

**Pull request closed without merging**